src/main/java/org/alfresco/rest/api/search/impl/ResultMapper.java

```diff
@@ -635,4 +642,31 @@ protected SolrJSONResultSet findSolrResultSet(ResultSet results)
 
         return null;
     }
+    public CollectionWithPagingInfo<TupleList> toCollectionWithPagingInfo(JSONArray docs, SearchSQLQuery searchQuery) throws JSONException
+    {
+        if(docs == null)
+        {
+            throw new RuntimeException("Solr response is required instead of docs:" + docs);
+        }
+        List<TupleList> entries = new ArrayList<TupleList>();
+        for(int i = 0; i < docs.length() -1; i++)
+        {
+            List<TupleEntry> row = new ArrayList<TupleEntry>();
+            JSONObject docObj = (JSONObject) docs.get(i);
+            docObj.keys().forEachRemaining(action -> {
+                try
+                {
+                    String value = docObj.get(action.toString()).toString();
+                    row.add(new TupleEntry(action.toString(), value));
+                } 
+                catch (JSONException e)
+                {
+                    throw new RuntimeException("Unable to parse SQL response. " + e);
+                }
+            });
+            entries.add(new TupleList(row));
+        }
+        Paging paging  = Paging.valueOf(0, 1000);
```

Comment:
Hard coded value of 1000? Were you maybe planning to get the paging value form searchQuery?

Reply:
no